go/arrow/array/string.go

```diff
@@ -113,6 +113,19 @@ func (a *String) setData(data *Data) {
 	if offsets := data.buffers[1]; offsets != nil {
 		a.offsets = arrow.Int32Traits.CastFromBytes(offsets.Bytes())
 	}
+
+	if a.array.data.length < 1 {
+		return
+	}
+
+	expNumOffsets := a.array.data.offset + a.array.data.length + 1
+	if len(a.offsets) < expNumOffsets {
+		panic("arrow/array: string offsets missing")
```

Comment:
let's add a little more detail on this error message:

`panic(fmt.Errorf("arrow/array: string offset buffer must have at least %d values", expNumOffsets))`